src/js/components/AppListFilterComponent.jsx

```diff
@@ -144,7 +132,7 @@ var AppListFilterComponent = React.createClass({
           onBlur={this.blurInputGroup}
           onChange={this.handleFilterTextChange}
           onFocus={this.focusInputGroup}
-          onKeyDown={this.handleKeyDown}
+          onKeyUp={this.handleSubmit}
```

Comment:
You will have a race condition here. If the response for the previously typed request comes after the new one, you will filter incorrectly. I suggest you keep a state of the current request and discard it if a new key was typed.

Reply:
I'm not sure to get your point. ( actually I get what you say ) but it seems to me that this is the whole point of react to handle your state for you 🤔 More than that, when we filter, we're not doing any request, we've already all the data locally.

Reply:
I was misled by the `handleSubmit` name which usually means “do the request”. No race condition if it’s synchronous and local then!